src/duke/Parser.java

```diff
@@ -0,0 +1,337 @@
+        try {
+            action = Commands.valueOf(obtainCommandWord());
+        } catch (IllegalArgumentException e) {
+            StringBuilder errorMessage = new StringBuilder();
+            errorMessage.append(line).append("☹ OOPS!!! I'm sorry, but I don't know what that means :-(\nTry another command!\n").append(line);
+            throw new IllegalArgumentException(errorMessage.toString());
+        }
+        switch (action) {
+            case todo:
+                System.out.println(createToDoTask());
+                break;
+            case deadline:
+                System.out.println(createDeadlineTask());
+                break;
+            case event:
+                System.out.println(createEventTask());
+                break;
+            case list:
+                System.out.println(listOutTasks());
+                break;
+            case mark:
+                System.out.println(markTaskAsDone());
+                break;
+            case unmark:
+                System.out.println(unmarkTask());
+                break;
+            case delete:
+                System.out.println(deleteTask());
+                break;
+            case find:
+                System.out.println(findTask());
+                break;
+            case bye:
+                System.out.println(terminateAndSaveProgram(storage));
+                System.exit(0);
+            default:
+        }
+    }
+
+    /**
+     * Makes a call to TaskList's printTasks()
+     */
+    public String listOutTasks() {
+        StringBuilder successMessage = new StringBuilder();
+        successMessage.append(line).append("Here are the tasks in your list: \n");
+        for (int i = 0; i < tasks.getTaskListSize(); i ++) {
+            successMessage.append(String.format("%o. ", i + 1));
+            successMessage.append(tasks.getParticularTask(i).toString());
+        }
+        successMessage.append(line);
+        return successMessage.toString();
+    }
+
+    public void checkTaskNumber() throws DukeException {
+        int taskNumber = obtainTaskNumber();
+        if (taskNumber >= tasks.getTaskListSize() || taskNumber < 1) {
+            throw new DukeException("Please provide the correct Task number! Maybe review the list of tasks first,\nAnd then execute the command for mark/unmark/delete!\n");
+        }
+    }
+
+    /**
+     * Makes a call on TaskList's mark()
+     */
+    public String markTaskAsDone() {
+        Task task;
+        StringBuilder successMessage = new StringBuilder();
+        successMessage.append(line).append("Nice! I've marked this task as done: \n");
+        try {
+            checkTaskNumber();
+            task = tasks.markTask(obtainTaskNumber());
+            successMessage.append(task.toString()).append(line);
+            return successMessage.toString();
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * Makes a call on duke.TaskList's unmark()
```

Comment:
I'm not too sure if this is descriptive enough. Let's say if this was another method and not an unmark task method, and the Javadocs said something like: `Makes a call on foo's bar()`. There isn't much description here, don't you think?